### PR TITLE
Dataset v2 openapi

### PR DESCRIPTION
**Pull request by sinclairs**
OpenAPI definition work to support V2 of dataset resource.

Some clunky bits in this PR to work around the JSON Schema / OpenAPI v3.0.2 schema mismatch issue. Happy to talk through these.

### CODE REVIEW

File: openapi/src/json-schemas/models/definitions/dataset/simple-open-question.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+            "const": "simple"
+        },
+        "id": {
+            "$ref": "../question-id.json"
+        },
+        "label": {
+            "type": "string"
+        },
+        "closedQuestion": {
+            "enum": [false]
```
Comment (armoj) [inline openapi/src/json-schemas/models/definitions/dataset/simple-open-question.json:16]: could we use `"const": false` here?

File: openapi/openapi.json [hunk 2]
```diff
@@ -2308,113 +2319,628 @@
...
+                                                                                                                                    {
+                                                                                                                                        "type": "array",
+                                                                                                                                        "items": {}
+                                                                                                                                    }
+                                                                                                                                ]
+                                                                                                                            }
+                                                                                                                        }
+                                                                                                                    },
```
Comment (armoj) [inline openapi/openapi.json:2730]: amazing formatting, thanks GH!!!

File: openapi/src/openapi-src.json [hunk 3]
```diff
@@ -600,7 +600,25 @@
                 "operationId": "getDataset",
                 "parameters": [
                     {
-                        "$ref": "#/components/parameters/questionnaireId"
+                        "name": "questionnaireId",
+                        "in": "path",
+                        "required": true,
+                        "description": "The id of the specific questionnaire instance. Format UUIDv4.",
```
Comment (armoj) [inline openapi/src/openapi-src.json:606]: what's the reasoning behind moving `questionnaireId` out of the `components` object? Could we not have both the `questionnaireId` and the `Accept-Version` in the `components` object for consistency?

**Review by armoj — CHANGES_REQUESTED** [review]

File: openapi/src/json-schemas/models/definitions/dataset/simple-open-question.json (hunk 1 above)
Comment (sinclairs) [reply openapi/src/json-schemas/models/definitions/dataset/simple-open-question.json:16]: This is a clunky bit... for some reason speccy won't allow `const` to have a boolean, so it needs to be an enum. (OpenAPI v3.0.2 doesn't support the keyword `const`)

File: openapi/src/openapi-src.json (hunk 3 above)
Comment (sinclairs) [reply openapi/src/openapi-src.json:606]: Another clunky bit... I had to being this inline to allow it to have a different `example` id. Speccy complained when I had it with as a `$ref` with a sibling `example` keyword.

As for the `Accept-Version` being inlined too... I did have this as a param definition, however the semvers are specific to the endpoint, so there wasn't much reusability to be had.

File: openapi/openapi.json (hunk 2 above)
Comment (sinclairs) [reply openapi/openapi.json:2730]: Erm, naughty bad GH 🤷‍♂️

File: openapi/src/json-schemas/models/definitions/dataset/simple-open-question.json (hunk 1 above)
Comment (armoj) [reply openapi/src/json-schemas/models/definitions/dataset/simple-open-question.json:16]: yeah thought so. just checking

File: openapi/src/openapi-src.json [hunk 4]
```diff
@@ -631,18 +649,163 @@
...
+                                    },
+                                    "v2.0.0": {
+                                        "summary": "Request Header \"Accept-Version: 2.0.0\"",
+                                        "value": {
+                                            "data": [
+                                                {
+                                                    "type": "dataset",
+                                                    "id": "0",
+                                                    "attributes": {
+                                                        "values": [
```
Comment (BarryPiccinni) [inline openapi/src/openapi-src.json:684]: I'm wondering if we're not using the term "value" too often? Consumers of this object are going to have to navigate paths like: 

```dataset.values.forEach(value => {
    let answer = value.value;
})```

Not an huge issue but maybe something to consider?

Comment (sinclairs) [reply openapi/src/openapi-src.json:684]: @BarryPiccinni the "[summary](https://github.com/CriminalInjuriesCompensationAuthority/data-capture-service/pull/224/files#diff-2836d18055d85134d1f93a05d9d4a74048ef60ddac08dee1ffb30d8b5c3c97f7R677)" and "[value](https://github.com/CriminalInjuriesCompensationAuthority/data-capture-service/pull/224/files#diff-2836d18055d85134d1f93a05d9d4a74048ef60ddac08dee1ffb30d8b5c3c97f7R678)" properties are part of the OpenAPI definition, not the actual response. Is that what you were referring to?

**Review by armoj — APPROVED** [review]

**Pull request merged**